pkg/aggregator/demultiplexer.go

```diff
@@ -776,7 +793,12 @@ func InitAndStartServerlessDemultiplexer(domainResolvers map[string]resolver.Dom
 	tagsStore := tags.NewStore(config.Datadog.GetBool("aggregator_use_tags_store"), "timesampler")
 
 	statsdSampler := NewTimeSampler(TimeSamplerID(0), bucketSize, tagsStore)
-	statsdWorker := newTimeSamplerWorker(statsdSampler, DefaultFlushInterval, bufferSize, metricSamplePool, flushAndSerializeInParallel{enabled: false}, tagsStore)
+	flushAndSerializeInParallel := flushAndSerializeInParallel{
+		enabled:     config.Datadog.GetBool("aggregator_flush_metrics_and_serialize_in_parallel") && serializer != nil && serializer.IsIterableSeriesSupported(),
+		bufferSize:  config.Datadog.GetInt("aggregator_flush_metrics_and_serialize_in_parallel_buffer_size"),
+		channelSize: config.Datadog.GetInt("aggregator_flush_metrics_and_serialize_in_parallel_chan_size"),
+	}
```

Comment:
I think the same code exists already in the aggregator (or is it in demultiplexer now?) init code. Would it make senes to factor it out into a helper?

Reply:
In order to reduce the code conflicts, I have done the code change in the next PR: https://github.com/DataDog/datadog-agent/pull/11133/commits/85e21ffb90f29f199310c3f024217c84ce5f8373.